### PR TITLE
Make bezier canvases HiDPI-aware.

### PR DESCRIPTION
**Pull request by mstange**
This makes them look much better on my HiDPI Mac.

### CODE REVIEW

File: components/Graphic.jsx [hunk 1]
```diff
@@ -68,11 +73,13 @@ var Graphic = React.createClass({
 
   componentDidMount: function() {
     var cvs = this.refs.canvas;
-    cvs.width = this.defaultWidth;
-    cvs.height = this.defaultHeight;
+    var dpr = window.devicePixelRatio || 1;
```
Comment (Pomax) [inline components/Graphic.jsx]: It's nicer not to have to repeat this evaluation in four different places, so let's instead give the Graphic class a function to retrieve this value:
```
getPixelRatio() {
  return window.devicePixelRatio || 1;
}
```
and then `var dpr = this.getPixelRatio();` in the rest of the code

File: components/Graphic.jsx [hunk 2]
```diff
@@ -279,12 +292,17 @@ var Graphic = React.createClass({
     var dataURL = this.refs.canvas.toDataURL();
     var img = new Image();
     img.src = dataURL;
+    var dpr = window.devicePixelRatio || 1;
```
Comment (Pomax) [inline components/Graphic.jsx]: we can turn this one into just `img.devicePixelRatio = this.getPixelRatio();`

File: components/Graphic.jsx [hunk 3]
```diff
@@ -279,12 +292,17 @@ var Graphic = React.createClass({
...
+    img.devicePixelRatio = dpr;
     return img;
   },
 
   setPanelCount: function(c) {
+    this.panelCount = c;
     var cvs = this.refs.canvas;
-    cvs.width = c * this.defaultWidth;
+    var dpr = window.devicePixelRatio || 1;
```
Comment (Pomax) [inline components/Graphic.jsx]: we can turn this one into just `cvs.width = c * this.defaultWidth * this.getPixelRatio();`

File: components/Graphic.jsx [hunk 4]
```diff
@@ -234,15 +241,21 @@ var Graphic = React.createClass({
...
     this.offset = {x:0, y:0};
     this.colorSeed = 0;
   },
 
   setSize: function(w,h) {
     this.defaultWidth = w;
     this.defaultHeight = h;
-    this.refs.canvas.width = w;
-    this.refs.canvas.height = h;
+    var dpr = window.devicePixelRatio || 1;
```
Comment (Pomax) [inline components/Graphic.jsx]: I'd like to reorganize the code here a little bit, just so there's some aliasing for used-more-than-once objects, and operations are grouped:

```
var cvs = this.refs.canvas;
var cstyle = cvs.style;
cstyle.width = this.panelCount * w + "px";
cstyle.height = h + "px";

var dpr = this.getPixelRatio();
cvs.width = this.panelCount * w * dpr;
cvs.height = h * dpr;
this.ctx.scale(dpr, dpr);
```

**Comment by mstange** [comment]
Thanks for the review! I've made the requested changes.

**Comment by Pomax** [comment]
Awesome, thanks! I'll merge this in and will generate an updated page later this evening.

**Pull request merged**